### PR TITLE
Split up README.md into separate files.

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Refactoring
- Documentation

### Description

REAME.md is getting a bit unwieldy, let's refactor it to help the user find what they need.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [x] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Review by franvila — APPROVED** [review]

File: DEV_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+```
+the send this input (including the empty line):
+```
+GET /version HTTP/1.1
+Host: www.example.com
+
+```
+
+You'll see an API response.  If the service_timeout change is effective, the socat
+will continue indefinitely.  If `socat` terminates after about 10 seconds, the workaround
```
Comment (SamBarker) [inline DEV_GUIDE.md:277]: Whoops looks like we missed a change 
```suggestion
will continue for 3 mins.  If `socat` terminates after about 10 seconds, the workaround
```

File: PERFORMANCE.md [hunk 2]
```diff
@@ -146,3 +146,15 @@ new ProduceRecordTransformationFilter(
 10000000 records sent, 60679.611650 records/sec (59.26 MB/sec), 25.00 ms avg latency, 1080.00 ms max latency, 1 ms 50th, 170 ms 95th, 601 ms 99th, 992 ms 99.9th.
 
 ```
+
+### Jenkins pipeline for performance
+When a PR is created and the performance tests are needed, the following comment shall be added into the PR:
```
Comment (SamBarker) [inline PERFORMANCE.md:151]: We should probably mention that this is restricted to specific people.

**Review by SamBarker — APPROVED** [review]
LGTM

File: PERFORMANCE.md (hunk 2 above)
Comment (k-wall) [reply PERFORMANCE.md:151]: +1 @franvila could you suggest some words.

Comment (franvila) [reply PERFORMANCE.md:151]: We could say something like "Only available for kroxylicious developers"

**Review by franvila — APPROVED** [review]
LGTM

File: PERFORMANCE.md (hunk 2 above)
Comment (k-wall) [reply PERFORMANCE.md]: @franvila how does that look?

Comment (k-wall) [reply PERFORMANCE.md]: I see you've looked!

**Pull request merged**